ext/gettext/gettext.c

```diff
@@ -61,6 +62,12 @@ ZEND_GET_MODULE(php_gettext)
 		RETURN_THROWS(); \
 	}
 
+#define PHP_DCGETTEXT_CATEGORY_CHECK(_arg_num, category) \
+	if (category == LC_ALL) { \
+		zend_argument_value_error(_arg_num, "is LC_ALL"); \
```

Comment:
The error message is not clear this way, so something along the lines of would be better IMO:
```suggestion
		zend_argument_value_error(_arg_num, "cannot be LC_ALL"); \
```